### ISSUE TITLE
Metastore failed to send event to control plane on startup

### ISSUE THREAD

**Issue by guilload** [quickwit-oss/quickwit#2686]
```
2023-01-20T19:49:07.301Z  INFO quickwit: version="0.4.0-nightly" commit="e2e18ae"
2023-01-20T19:49:07.310Z  INFO quickwit_config::quickwit_config::serialize: Using listen address as advertise address. advertise_address=127.0.0.1
2023-01-20T19:49:07.314Z  WARN quickwit_config::quickwit_config::serialize: Cluster ID is not set, falling back to default value. cluster_id=quickwit-default-cluster
2023-01-20T19:49:07.314Z  WARN quickwit_config::quickwit_config::serialize: Peer seed list is empty.
2023-01-20T19:49:07.314Z  INFO quickwit_cli: Loaded Quickwit config. config_uri=file:///Users/guilload/Workspace/repositories/quickwit/config/quickwit.yaml config=QuickwitConfig { cluster_id: "quickwit-default-cluster", node_id: "node-lingering-jYZT", enabled_services: {Metastore, Janitor, ControlPlane, Indexer, Searcher}, rest_listen_addr: 127.0.0.1:7280, gossip_listen_addr: 127.0.0.1:7280, grpc_listen_addr: 127.0.0.1:7281, gossip_advertise_addr: 127.0.0.1:7280, grpc_advertise_addr: 127.0.0.1:7281, peer_seeds: [], data_dir_path: "/Users/guilload/Workspace/repositories/quickwit/qwdata", metastore_uri: Uri { uri: "file:///Users/guilload/Workspace/repositories/quickwit/qwdata/indexes#polling_interval=30s" }, default_index_root_uri: Uri { uri: "file:///Users/guilload/Workspace/repositories/quickwit/qwdata/indexes" }, indexer_config: IndexerConfig { enable_opentelemetry_otlp_service: true, split_store_max_num_bytes: Byte(100000000000), split_store_max_num_splits: 1000, max_concurrent_split_uploads: 12 }, searcher_config: SearcherConfig { enable_jaeger_service: true, fast_field_cache_capacity: Byte(1000000000), split_footer_cache_capacity: Byte(500000000), max_num_concurrent_split_searches: 100, max_num_concurrent_split_streams: 100 }, ingest_api_config: IngestApiConfig { max_queue_memory_usage: 2147483648, max_queue_disk_usage: 4294967296 } }
2023-01-20T19:49:07.315Z  INFO quickwit_cluster::cluster: Joining cluster. cluster_id=quickwit-default-cluster node_id=node-lingering-jYZT enabled_services={Metastore, Janitor, ControlPlane, Indexer, Searcher} gossip_listen_addr=127.0.0.1:7280 gossip_advertise_addr=127.0.0.1:7280 grpc_advertise_addr=127.0.0.1:7281 peer_seed_addrs=
2023-01-20T19:49:07.316Z  INFO chitchat::server: initial_seed_addrs={}
2023-01-20T19:49:07.321Z  INFO quickwit_grpc_clients::balance_channel: Adding `control_plane` servers to client pool. server_addresses={127.0.0.1:7281}
2023-01-20T19:49:07.348Z  INFO quickwit_ingest_api::ingest_api_service: Ingest API partition id ingest_partition_id=ingest_partition_01GQ51VEDQDCFQ6TJR0VWT83NV
2023-01-20T19:49:07.352Z  INFO quickwit_config::index_config::serialize: Index config does not specify `index_uri`, falling back to default value. index_id=otel-trace-v0 index_uri=file:///Users/guilload/Workspace/repositories/quickwit/qwdata/indexes/otel-trace-v0
2023-01-20T19:49:07.424Z ERROR quickwit_metastore::metastore::metastore_with_control_plane_triggers: Failed to send index event to the control plane on `create-source`: `status: Unavailable, message: "error trying to connect: tcp connect error: Connection refused (os error 61)", details: [], metadata: MetadataMap { headers: {} }`.
2023-01-20T19:49:07.435Z  INFO quickwit_indexing: Starting indexer service.
2023-01-20T19:49:07.444Z  INFO quickwit_indexing::actors::ingest_api_garbage_collector: ingest-api-garbage-collect-operation
2023-01-20T19:49:07.445Z  INFO quickwit_janitor: Starting janitor service.
2023-01-20T19:49:07.446Z  INFO quickwit_janitor::actors::garbage_collector: garbage-collect-operation
2023-01-20T19:49:07.446Z  INFO quickwit_janitor::actors::garbage_collector: Garbage collecting indexes. index_ids=otel-trace-v0
2023-01-20T19:49:07.448Z  INFO quickwit_janitor::actors::delete_task_pipeline: Spawning delete tasks pipeline. index_id=otel-trace-v0 root_dir=/Users/guilload/Workspace/repositories/quickwit/qwdata/delete_task_service
2023-01-20T19:49:07.448Z  INFO quickwit_control_plane::scheduler: Apply physical indexing plan: PhysicalIndexingPlan { indexing_tasks_per_node_id: {"node-lingering-jYZT": [IndexingTask { index_id: "otel-trace-v0", source_id: "_ingest-api-source" }]} }
2023-01-20T19:49:07.448Z ERROR quickwit_control_plane::scheduler: Cannot create indexing service client for indexer, skip indexing plan. indexer_node_id=node-lingering-jYZT err=transport error

Caused by:
    0: error trying to connect: tcp connect error: Connection refused (os error 61)
    1: tcp connect error: Connection refused (os error 61)
    2: Connection refused (os error 61)
```

**Comment by guilload** [comment]
Indexers going up and down are also expected. In this case, the indexer was probably not ready yet. The control plane should retry and maybe log a warning instead?

**Comment by guilload** [comment]
I ran into this issue again. The issue is that nodes declare themselves ready (i.e. they have established a connection to the metastore) before actually starting their services (gRPC services and REST handlers). We need to spawn the readiness task last.

**Comment by guilload** [comment]
Closed via #2940.
